src/createTokenBridge.integration.test.ts

```diff
@@ -0,0 +1,193 @@
+      );
+
+      const sequencerConfigJson = JSON.parse(
+        execSync('docker exec ' + container + ' cat /config/sequencer_config.json').toString(),
+      );
+
+      return {
+        rollup: deploymentJson['rollup'],
+        // this works because rollup owner, batch poster is all the same
+        //
+        // https://github.com/OffchainLabs/nitro-testnode/blob/master/test-node.bash#L363
+        rollupOwner: sequencerConfigJson['node']['batch-poster']['parent-chain-wallet']['account'],
+      };
+    } catch {
+      // empty on purpose
+    }
+  }
+
+  throw new Error('nitro-testnode sequencer not found');
+}
+
+const deployer = getTestPrivateKeyAccount();
+
+const nitroTestnodeL1Client = createPublicClient({
+  chain: nitroTestnodeL1,
+  transport: http(nitroTestnodeL1.rpcUrls.default.http[0]),
+});
+
+const nitroTestnodeL2Client = createPublicClient({
+  chain: nitroTestnodeL2,
+  transport: http(nitroTestnodeL2.rpcUrls.default.http[0]),
+});
+
+it(`successfully deploys token bridge contracts through token bridge creator`, async () => {
+  const { rollup, rollupOwner } = getRollupData();
+
+  // deploy a fresh token bridge creator, because it is only possible to deploy one token bridge per rollup per token bridge creator
+  const tokenBridgeCreator = await deployTokenBridgeCreator({
+    publicClient: nitroTestnodeL1Client,
+  });
+
+  const txRequest = await createTokenBridgePrepareTransactionRequest({
+    params: {
+      rollup,
+      rollupOwner,
+    },
+    parentChainPublicClient: nitroTestnodeL1Client,
+    childChainPublicClient: nitroTestnodeL2Client,
+    account: deployer.address,
+  });
+
+  // update the transaction request to use the fresh token bridge creator
+  const txRequestToFreshTokenBridgeCreator = { ...txRequest, to: tokenBridgeCreator };
+
+  // sign and send the transaction
+  const txHash = await nitroTestnodeL1Client.sendRawTransaction({
+    serializedTransaction: await deployer.signTransaction(txRequestToFreshTokenBridgeCreator),
+  });
+
+  // get the transaction receipt after waiting for the transaction to complete
+  const txReceipt = createTokenBridgePrepareTransactionReceipt(
+    await nitroTestnodeL1Client.waitForTransactionReceipt({ hash: txHash }),
+  );
+
+  function waitForRetryables() {
+    return txReceipt.waitForRetryables({ orbitPublicClient: nitroTestnodeL2Client });
+  }
+
+  expect(txReceipt.status).toEqual('success');
+  // assert promise is resolved with 2 retryables
+  await expect(waitForRetryables()).resolves.toHaveLength(2);
+
+  // get contracts
+  const coreContracts = await createRollupFetchCoreContracts({
+    rollup,
+    publicClient: nitroTestnodeL1Client,
+  });
+
+  const tokenBridgeContracts = await createTokenBridgeFetchTokenBridgeContracts({
+    inbox: coreContracts.inbox,
+    parentChainPublicClient: nitroTestnodeL1Client,
+  });
+
+  // check parent chain contracts
+
+  // parent chain router
+  const [
```

Comment:
We can add these tests that follow the same logic as [this script of the token-bridge-repo](https://github.com/OffchainLabs/token-bridge-contracts/blob/v1.2.0/test-e2e/tokenBridgeDeploymentTest.ts).

What do you think @spsjvc ? (Didn't want to add all of them until having confirmation from you first)

Reply:
Removed these tests as they should are already part of the test suite of token-bridge-repo. Replaced them with tests about the token bridge contracts fetched (number of contracts and non-zero addresses)